client/db/src/lib.rs

```diff
@@ -521,6 +625,58 @@ impl<Block: BlockT> BlockchainDb<Block> {
 		let mut meta = self.meta.write();
 		meta.block_gap = gap;
 	}
+
+	/// Empty the cache of pinned items.
+	fn clear_pinning_cache(&self) {
+		self.pinned_blocks_cache.clear();
+	}
+
+	/// Load a justification into the cache of pinned items.
+	/// Reference count of the item will not be increased. Use this
+	/// to load values for items items into the cache which have already been pinned.
+	fn insert_justifications_if_pinned(&self, hash: Block::Hash, justification: Justification) {
+		if !self.pinned_blocks_cache.contains(&hash) {
```

Comment:
`contains` releases the lock, so the cache can be modified between call to `contains` and `insert_justification`. Is it on purpose?